addons/test_mail/tests/test_attachment.py

```diff
@@ -0,0 +1,72 @@
+from odoo.addons.test_mail.tests.common import TestMailCommon
+from odoo.exceptions import AccessError
+
+class TestAttachment(TestMailCommon):
+
+    def test_attachment_forbid_unlink(self):
+        """Check that removing message attachments is prevented on other user's messages."""
+        def write(attachment, w_user, w_sudo=False):
+            attachment.with_user(w_user).sudo(w_sudo).write({'name': f'{w_user.name} {attachment.name[-1]}'})
+
+        def unlink(attachment, w_user, w_sudo=False):
+            attachment.exists().with_user(w_user).sudo(w_sudo).unlink()
+
+        test_record = self.env['mail.test.simple'].with_context(self._test_context).create({
+            'name': 'Test',
+            'email_from': 'ignasse@example.com',
+        })
+
+        user_admin, user_employee = self.env['res.users'].search([('login', 'in', ['admin', 'employee'])], order='login asc')
+        admin_attachments = self.env['ir.attachment'].with_user(user_admin).create([{
+            'type': 'binary',
+            'name': f'admin_attachment_{n}',
+            'datas': "ABC=",
+        } for n in range(2)])
+        employee_attachments = self.env['ir.attachment'].with_user(user_employee).create([{
+            'type': 'binary',
+            'name': f'employee_attachment_{n}',
+            'datas': "ABC=",
+        } for n in range(3)])
+
+        user_employee_clone = user_employee.copy(default={'login': 'employee_clone', 'email': 'employee_clone@local.lan'})
```

Comment:
Could use ``mail_new_test_user`` (not sure the "copy" is required, is it ? it is a bit intriguing ^^ )

Reply:
Not required but it's an easy way to get a known valid user without having to specify anything.
Will go with `mail_new_test_user` for set up clarity though